telematic_system/telematic_apps/web_app/server/file_upload/s3_list_objects.js

```diff
@@ -27,31 +27,37 @@ const secretAccessKey = process.env.AWS_SECRET_KEY;
 const region = process.env.S3_REGION;
 const bucket = process.env.S3_BUCKET;
 
-exports.listObjects = async () => {
+exports.listObjects = async (s3Folder) => {
   const client = new S3Client({
     credentials: {
       accessKeyId,
       secretAccessKey,
     },
     region,
   });
-  const command = new ListObjectsV2Command({
-    Bucket: bucket,
-  });
   let contents = [];
-  let isTruncated = true;
-  while (isTruncated) {
-    const { Contents, IsTruncated, NextContinuationToken } = await client.send(
-      command
-    );
-    const contentsList = Contents.map((c) => ({
-      original_filename: c.Key,
-      size: c.Size,
-      filepath: bucket,
-    }));
-    isTruncated = IsTruncated;
-    contents.push(...contentsList);
-    command.input.ContinuationToken = NextContinuationToken;
+  try {
+    const command = new ListObjectsV2Command({
+      Bucket: bucket,
+      Prefix: s3Folder + '/',
+      Delimiter: '/',
+    });
+    let isTruncated = true;
+    while (isTruncated) {
+      const { Contents, IsTruncated, NextContinuationToken } = await client.send(
+        command
+      );
+      const contentsList = Contents.map((c) => ({
+        original_filename: c.Key,
+        size: c.Size,
+        filepath: bucket,
+      }));
+      isTruncated = IsTruncated;
+      contents.push(...contentsList);
+      command.input.ContinuationToken = NextContinuationToken;
+    }
+  } catch (err) {
+    console.error("Cannot find files in S3 bucket: " + bucket + ", folder:  "+ s3Folder)
```

Comment:
Is this error statement correct? Should it be cannot upload files?

Reply:
This is returning a list of existing files in AWS S3 bucket in support of uploading to S3 through CLI, AWS management console or other means rather than telematic UI.